boards/DCM/Test/Inc/Test_Dcm.h

```diff
@@ -2,3 +2,11 @@
 
 #include <fff.h>
 #include <gtest/gtest.h>
+
+template <typename T>
+void TearDownObject(T *&object, void (*destroy_func)(T *object))
+{
+    ASSERT_TRUE(object != NULL);
+    destroy_func(object);
+    object = NULL;
+}
```

Comment:
Is this duplicated with the other boards?

Reply:
I thought about only defining this function once and sharing it across the other boards but wasn't sure where to put it. The most obvious choice seems to be creating a `Test_Shared.h`, but I felt like that was up for debate because the test files under `shared` are sort of standalone. That is, we create a target named `SHARED_test` that can be compiled on its own without other `BOARD_test` targets. I felt like the abstraction would become kind of weird if I start including `Test_Shared.h` in all the `BOARD_test` targets like:

```
// Test_Shared.h
void TearDownObject(...) {}
// Test_Bms.h
#include "Test_Shared.h" // Does this include make sense if we
                                         // can compile SHARED_test on its own, 
                                         // independent of BMS?
```

Maybe i'm just overthinking it, but that's why i defined the function multiple times.

Reply:
I would make a folder called `shared/test_utils` and put it in there to make it clear that it's not part of the shared tests, it's just shared testing code that anything (including the shared tests) can use.

Reply:
I like it, nice and simple. Added it in !